src/ChainSafe.Gaming/Web3/Evm/Wallet/WalletTransactionExecutor.cs

```diff
@@ -27,10 +28,20 @@ public WalletTransactionExecutor(IWalletProvider walletProvider, ISigner signer)
             this.signer = signer;
         }
 
-        public async Task<TransactionResponse> SendTransaction(TransactionRequest transaction)
+        public async Task<TransactionResponse> SendTransaction(TransactionRequest transaction, IGasFeeModifier gasFeeModifier = null)
         {
             transaction.From ??= signer.PublicAddress;
             transaction.Data ??= "0x";
+            gasFeeModifier ??= new NoGasFeeModifier();
```

Comment:
Idk if we need this, wallet handle these independently I think

Reply:
Ahhhh you are right, but then gasFeeModifier will not be used leading to a nullref down below. I was banging my head on that 😆

Reply:
I'm not an expert, but I think wallets do pick the gas price for you. Some of them (all?) even allow to set the gas price yourself. If it's the case, this functionality should still be useful for presetting gas fee strategies by developers for specific transactions, I think.

To be more confident we could just test out how this works with different wallets.

Another question is where does the 'initial' value of `transaction.GasPrice` and `transaction.MaxFeePerGas` comes from. Does we calculate/fetch it ourself somewhere?